Describe stat level in new character window

RenechCDDA opened the pull request:
#### Summary
Interface "Describe stat level in new character window"

#### Purpose of change
Many people ask "What does 8 strength mean?" or "Why can't I raise my strength once I get ingame?". We have those sorts of answers scattered around the documentation, but we could make it clearer ingame too.

#### Describe the solution
Describe the stat level in the new character window. This provides another 'soft' suggestion for what stat level you should (not) select. It also helps to show people why massively increasing their stats at the start is unbalanced, besides just kicking around their lifestyle score.

#### Describe alternatives you've considered
Repeating the same translator comment over and over again is kind of awkward, I could probably format this static function differently.

#### Testing
![image](https://github.com/user-attachments/assets/13cfc3ca-b10b-442c-aef4-a114256261e0)


#### Additional context

RenechCDDA committed:
Describe stat level in new character tab

Maleclypse commented:
What about impaired instead of crippled and enfeebled instead of legally disabled?  That also covers the facts that the PC can be impacted by stat changes that will change?

RenechCDDA committed:
Add another breakpoint on the high end for a more reasonable set of descriptions.

Just for formatting/clarity's sake, the range for descriptions is:

0-4: "legally disabled"
5-6: "crippled"
7: "below average"
8: "average human"
9-10: "above average"
11-12: "top 10%"
13-14: "top 1%"
15-19: "olympian"
20+: "superhuman"

RenechCDDA commented:
> What about impaired instead of crippled and enfeebled instead of legally disabled? That also covers the facts that the PC can be impacted by stat changes that will change?

I do like `impaired` more than `crippled`, it feels more neutral. ('Cripple' can come with some level of stigmatization)

I think 'enfeebled' for the lowest level is a bit too... wordy/esoteric though, do you have any other suggestions?

(Haven't pushed for this yet, because I want to get a consensus)


Also note that this ONLY displays during the new character window, e.g. looking at the stat in the ingame status/@ menu does not  show it. (There is not enough space there)

Holli-Git commented on src/newcharacter.cpp:
```diff
@@ -1457,6 +1457,42 @@ static std::string assemble_stat_help( const input_context &ctxt )
+        description = _( "superhuman" );
+    } else if( stat_value > 14 ) {
+        //~Description of a character's main stats. Should not exceed 17 characters of width.
+        description = _( "olympian" );
```
Maybe instead of "olympian" it could be "worldclass"? Feels similar enough but still distinct

RenechCDDA committed:
Shuffle the stat description around for more space

RenechCDDA committed:
Some suggested changes to the description phrases

akrieger commented:
Incapacitated? Incapable? Infirm?

Maleclypse committed:
Update src/newcharacter.cpp

kevingranade commented on src/newcharacter.cpp:
```diff
@@ -1457,6 +1457,42 @@ static std::string assemble_stat_help( const input_context &ctxt )
+        description = _( "impaired" );
+    } else if( stat_value >= 0 ) {
+        //~Description of a character's main stats. Should not exceed 18 characters of width.
+        description = _( "legally disabled" );
```
```suggestion
        description = _( "incapacitated" );
```

akrieger committed:
Update src/newcharacter.cpp

Co-authored-by: Kevin Granade <kevin.granade@gmail.com>
